java-manta-client-unshaded/src/main/java/com/joyent/manta/config/ConfigContext.java

```diff
@@ -141,6 +141,12 @@ default String getMantaHomeDirectory() {
      */
     Integer getSkipDirectoryDepth();
 
+    /**
+     * @return the maximum amount of empty parent directories to delete. -1 will delete all empty parent directories 0
+     *         will delete none of them.
+     */
+    Integer getPruneEmptyParentDepth();
```

Comment:
I can only add comments within changed areas but this setting should also be checked in the `validate` method. Search for "Manta skip directory depth must be 0 or greater" (should be around line 344 according to this diff)